ratpack-pac4j/src/main/java/ratpack/pac4j/internal/Pac4jAuthenticator.java

```diff
@@ -81,13 +82,28 @@ public void handle(Context ctx) throws Exception {
   }
 
   public Clients createClients(Context ctx, PathBinding pathBinding) throws Exception {
+    String callback = createCallbackUrl(ctx.get(PublicAddress.class), pathBinding, path);
+
+    return new Clients(callback, getClientsFromProvider(ctx, clientsProvider));
+  }
+
+  private String createCallbackUrl(PublicAddress publicAddress, PathBinding pathBinding, String path) {
     String boundTo = pathBinding.getBoundTo();
-    PublicAddress publicAddress = ctx.get(PublicAddress.class);
-    String absoluteCallbackUrl = publicAddress.get() + boundTo + "/" + path;
+    URI address = publicAddress.get();
+
+    if(!boundTo.isEmpty()) {
+      if (!(address.toString().endsWith("/") || boundTo.startsWith("/"))) {
```

Comment:
I see, you have a point about the first part of the condition, wrt public address but I'm not sure we should be this defensive here, what do you think @alkemist? On the other hand I'm sure that the second part of the condition can go because  bound path binding will never start with a `/`.